python/ray/serve/multiplex.py

```diff
@@ -0,0 +1,90 @@
+from ray._private.async_compat import sync_to_async
+from collections import OrderedDict
+from typing import Any, Callable
+import logging
+from ray.serve._private.constants import SERVE_LOGGER_NAME
+import inspect
+import asyncio
+
+
+logger = logging.getLogger(SERVE_LOGGER_NAME)
+
+
+class _ModelMultiplexWrapper:
+    def __init__(
+        self,
+        model_load_func: Callable[[str], Any],
+        self_arg: Any,
+        max_num_models_per_replica: int,
+    ):
+        """Initialize the model multiplexer.
+
+        The model multiplexer is a wrapper class that wraps the model load function
+        and provides the LRU caching functionality, and the model load function should
+        be a coroutine function that takes the model ID as the first argument and
+        returns the model handle.
+        The model multiplexer will also ensure that the number of models on the current
+        replica does not exceed the specified limit.
+        The model will be unloaded in the LRU order, the model multiplexer will call the
+        model's __del__ attribute if it exists to clean up the model resources eagerly.
```

Comment:
summary of the class should go in class-level docstring, this should be documenting the behavior of the constructor

see: https://google.github.io/styleguide/pyguide.html#384-classes